pkg/restore/restore.go

```diff
@@ -634,16 +635,29 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		}
 
 		// necessary because we may have remapped the namespace
-		obj.SetNamespace(namespace)
+		// if the namespace is blank, don't create the key
+		if namespace != "" {
+			obj.SetNamespace(namespace)
+		}
 
 		// add an ark-restore label to each resource for easy ID
 		addLabel(obj, api.RestoreLabelKey, ctx.restore.Name)
 
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
 		_, err = resourceClient.Create(obj)
 		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
-			continue
+			fromCluster, _ := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+			// Remove insubstantial metadata
+			fromCluster, _ = resetMetadataAndStatus(fromCluster)
+			// Add the restore's label to the in-cluster object since we know that's different
+			addLabel(fromCluster, api.RestoreLabelKey, ctx.restore.Name)
+
+			if equality.Semantic.DeepEqual(obj, fromCluster) {
+				continue
```

Comment:
probably makes sense to do a `ctx.infof(...)` here stating that the object already exists with matching state in-cluster